src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -133,8 +139,8 @@ internal abstract class BrokerClient : IBrokerClient {
                 } catch (UnauthorizedAccessException) {
                     isValidCredentials = false;
                     continue;
-                } catch (HttpRequestException ex) {
-                    throw new RHostDisconnectedException("HTTP error while creating session: " + ex.Message, ex);
+                } catch (BrokerApiErrorException apiex) {
+                    throw new RHostDisconnectedException(apiex.ApiError);
```

Comment:
Yeah, there are many places that create default but fair amount that supply custom message which then sometimes displayed in UI. There are 3 places that provide both message AND inner exception. This should be untangled and simplified, but will take some effort to straighten it out.

Basically, it is safe to provide the message. Alternatively we could supply both message and the inner exception. Supplying just the inner part may break code that expects readable message for the UI.